consensus/double_sign.go

```diff
@@ -12,96 +12,111 @@ import (
 // Returns true when it is a double-sign or there is error, otherwise, false.
 func (consensus *Consensus) checkDoubleSign(recvMsg *FBFTMessage) bool {
 	if consensus.couldThisBeADoubleSigner(recvMsg) {
-		if alreadyCastBallot := consensus.Decider.ReadBallot(
-			quorum.Commit, recvMsg.SenderPubkey.Bytes,
-		); alreadyCastBallot != nil {
-			firstPubKey, err := bls.BytesToBLSPublicKey(alreadyCastBallot.SignerPubKey[:])
-			if err != nil {
-				return false
-			}
-			if recvMsg.SenderPubkey.Object.IsEqual(firstPubKey) {
-				for _, blk := range consensus.FBFTLog.GetBlocksByNumber(recvMsg.BlockNum) {
-					firstSignedBlock := blk.Header()
-					areHeightsEqual := firstSignedBlock.Number().Uint64() == recvMsg.BlockNum
-					areViewIDsEqual := firstSignedBlock.ViewID().Uint64() == recvMsg.ViewID
-					areHeadersEqual := firstSignedBlock.Hash() == recvMsg.BlockHash
+		addrSet := map[common.Address]struct{}{}
+		for _, pubKey2 := range recvMsg.SenderPubkeys {
+			if alreadyCastBallot := consensus.Decider.ReadBallot(
+				quorum.Commit, pubKey2.Bytes,
+			); alreadyCastBallot != nil {
+				for _, pubKey1 := range alreadyCastBallot.SignerPubKeys {
+					firstPubKey, err := bls.BytesToBLSPublicKey(pubKey1[:])
```

Comment:
can we simply compare the bytes of pubKey1 with pubKey2.Bytes and skip, instead of `BytesToBLSPublicKey` and then skipping?

Reply:
Yup, definitely. done.